### ISSUE TITLE
Set variant.broken=true after errors in grade()

### ISSUE THREAD

**Issue by mwest1066** [PrairieLearn/PrairieLearn#5498]
If `grade()` throws a fatal error, we currently set `submission.broken = true` but we do _not_ set `variant.broken = true`. This means that errors in grading code can block students from ever progressing (e.g., if an element has been upgraded and no longer works with old data).

To fix this, we could set `variant.broken = true` in this case. We should check that this has a reasonable UI flow and also that it works correctly with `singleVariant` questions.

**Comment by nwalters512** [comment]
I don't know if this is a good idea. My primary concern here is transient issues, e.g. timeouts under heavy load or because question authors (against our advice) decide to make grading dependent on things like network calls. If grading fails in one of these ways, it would be completely natural for a student to retry, but this change would prevent them from doing so and force them to restart with a new variant.

**Comment by reteps** [comment]
Maybe we need to give students both options? I can make the argument the other way, that some randomized variants will break when students try to grade, and they should be able to get a new one during their exam. E.g. a random parameter leads to a division-by-zero error.

**Comment by reteps** [comment]
Matt and I discussed this on 2/16, we think that we could do this in multi-variant question scenarios: e.g. not marked `singleVariant`, and not on an exam (because then only one variant is completed by the student).

Matt was thinking that if there is an issue in a single-variant `grade()` cycle, e.g. on an exam, the student will have already spent significant time on the problem, and is already only expected to solve it once. To ask the student to solve it again because of a problem due to the instructor is unfair.

**Comment by nwalters512** [comment]
That seems reasonable, but we should think about how this behaves in a world where the Exam/Homework dichotomy doesn't exist.

**Comment by reteps** [comment]
I think we can derive "is this single variant" regardless of the exam/homework dichotomy

**Comment by mwest1066** [comment]
Yep, it's more about "single-variant" questions than the exam/homework type.
